sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java

```diff
@@ -232,20 +233,25 @@ private static void ensureRoutingMapPresent(
                 // upper logic will refresh cache and retry.
                 logger.debug("Collection cache or routing map cache is potentially outdated." +
                                  " Returning null. Upper logic will refresh cache and retry.");
-                return Mono.empty();
+                return Mono.just(new Utils.ValueHolder<>(null));
             }
 
-            Mono<AddressInformation[]> addressesObs = this.addressCache.tryGetAddresses(
+            Mono<Utils.ValueHolder<AddressInformation[]>> addressesObs = this.addressCache.tryGetAddresses(
                 request,
                 new PartitionKeyRangeIdentity(collection.getResourceId(), range.getId()),
                 forceRefreshPartitionAddresses);
 
-            return addressesObs.flatMap(addresses -> Mono.just(new ResolutionResult(range, addresses))).switchIfEmpty(Mono.defer(() -> {
-                logger.info(
+            return addressesObs.flatMap(addressesValueHolder -> {
+
+                if (addressesValueHolder.v == null) {
+                    logger.info(
                         "Could not resolve addresses for identity {}/{}. Potentially collection cache or routing map cache is outdated. Return empty - upper logic will refresh and retry. ",
```

Comment:
update the log message accordingly. "Return null" instead of "return empty"

Reply:
Will fix this in next iteration. Thanks for catching this.